### PR TITLE
[RHPAM-3888] SSO integration fails for multiple Realm certificates

### PR DESCRIPTION
**Pull request by desmax74**
See: https://issues.redhat.com/browse/RHPAM-3888

### CODE REVIEW

File: jboss/container/wildfly/launch/keycloak/added/keycloak.sh [hunk 1]
```diff
@@ -641,6 +641,22 @@ function configure_security_domain() {
   sed -i "s|<!-- ##KEYCLOAK_SECURITY_DOMAIN## -->|${keycloak_security_domain%$'\n'}|" "${CONFIG_FILE}"
 }
 
+function configure_realms_certificates(){
+   realms_certificates=$1
+   # We need it to be retrieved prior to iterate the web deployments, needed by CLI
+   if [ -n "$realms_certificates" ]; then
+      if [ -n "$token" ]; then
+        # SSO Server 7.0
+        realms_certificates=`$CURL -H "Accept: application/json" -H "Authorization: Bearer ${token}" ${sso_service}/admin/realms/${SSO_REALM} | $(jq '.keys[].certificate') `
```
Comment (jfdenise) [inline jboss/container/wildfly/launch/keycloak/added/keycloak.sh:650]: @desmax74 , jq command line is not installed in the EAP image. I suspect that it is installed in RHPAM image right?

**Comment by desmax74** [comment]
@jfdenise reverted the changes and applied only the jq command to retrieve the certificate with use SIG

File: jboss/container/wildfly/launch/keycloak/added/keycloak.sh [hunk 2]
```diff
@@ -668,8 +668,10 @@ function configure_subsystem() {
     # SSO Server 7.0
     realm_certificate=`$CURL -H "Accept: application/json" -H "Authorization: Bearer ${token}" ${sso_service}/admin/realms/${SSO_REALM} | grep -Po '(?<="certificate":")[^"]*'`
     if [ -z "$realm_certificate" ]; then
-      #SSO Server 7.1
-      realm_certificate=`$CURL -H "Accept: application/json" -H "Authorization: Bearer ${token}" ${sso_service}/admin/realms/${SSO_REALM}/keys | grep -Po '(?<="certificate":")[^"]*'`
+      #SSO Server 7.1 and newer. If is 7.5 we skip certificate with use=ENC
+      realm_certificate=`$CURL -H "Accept: application/json" -H "Authorization: Bearer ${token}" ${sso_service}/admin/realms/${SSO_REALM}/keys | $(jq '.keys[] | select( .certificate != null and .use == "SIG")') | $(jq '.certificate') `
```
Comment (jfdenise) [inline jboss/container/wildfly/launch/keycloak/added/keycloak.sh:672]: This change introduce a new dependency on jq command line. 
Is this fix: https://github.com/jboss-container-images/redhat-sso-7-openshift-image/pull/192/commits/214fca4461236a6430bf21c1f2e53fbcb53fec81
possibly a better candidate (no use of jq)? It covers 7.0, 7.1 to 7.4 and 7.5.

Comment (desmax74) [reply jboss/container/wildfly/launch/keycloak/added/keycloak.sh]: @jfdenise fix applied on the script

**Review by jfdenise — APPROVED** [review]
@desmax74 Thank-you. The fix looks good to me. We will handle porting this change to other branches.

**Review by luck3y — APPROVED** [review]

**Comment by spolti** [comment]
Hi @jfdenise, @luck3y thanks for approving it.
Do you think we can have this merged this week or next and a new tag from 0.18.x created?

**Comment by luck3y** [comment]
@spolti That sounds fine to me -- @jfdenise do you want me to open a version of this against main as well?

**Comment by jfdenise** [comment]
@spolti we are waiting confirmation from @desmax74 that the new implementation fix (patch shared with keycloak team) actually works for his use-case, then we can merge. @luck3y thank-you for proposing, go ahead for master (23.x too perhaps?). I will handle it for v2 (some PR are currently opened and I will need to integrate the fix too there).

**Comment by spolti** [comment]
Thanks @jfdenise, about the cherry-pick, if we can also have it on 23.x, we are planning to move to this branch in a near future
@desmax74 please let us know about the tests.

**Comment by desmax74** [comment]
@spolti @jfdenise @luck3y Green light on our QE tests with SSO 7.4 and SSO 7.5

**Comment by jfdenise** [comment]
@desmax74 Thank-you! Merged in 0.18.x

**Pull request merged**